tests/test_process.py

```diff
@@ -1,67 +1,242 @@
+    ) as patch_get_children:
+        df = process.create_data_manifest(parent="syn1111111", syn=syn)
+        patch_get_children.assert_called_once_with("syn1111111")
+        assert isinstance(df, pd.DataFrame)
+
+
+def test_process_all_files_config_path(syn):
+    with patch.object(
+        utils,
+        "_get_config",
+        return_value=[{"destination": "destination"}, {"datasets": ["a", "b", "c"]}],
+    ) as patch_get_config, patch.object(
+        load, "create_temp_location", return_value=None
+    ), patch.object(
+        process, "process_dataset", return_value=tuple()
+    ), patch.object(
+        process,
+        "create_data_manifest",
+        return_value=pd.DataFrame({"id": ["a", "b", "c"]}),
+    ), patch.object(
+        load, "df_to_csv", return_value="path/to/csv"
+    ), patch.object(
+        load, "load", return_value=None
+    ):
+        process.process_all_files(config_path="path/to/config", syn=syn)
+        patch_get_config.assert_called_once_with(config_path="path/to/config")
+
+
+def test_process_all_files_no_config_path(syn):
+    with patch.object(
+        utils,
+        "_get_config",
+        return_value=[{"destination": "destination"}, {"datasets": ["a", "b", "c"]}],
+    ) as patch_get_config, patch.object(
+        load, "create_temp_location", return_value=None
+    ), patch.object(
+        process, "process_dataset", return_value=tuple()
+    ), patch.object(
+        process,
+        "create_data_manifest",
+        return_value=pd.DataFrame({"id": ["a", "b", "c"]}),
+    ), patch.object(
+        load, "df_to_csv", return_value="path/to/csv"
+    ), patch.object(
+        load, "load", return_value=None
+    ):
+        process.process_all_files(config_path=None, syn=syn)
+        patch_get_config.assert_called_once_with()
+
+
+def test_process_all_files_full(syn):
+    with patch.object(
+        utils,
+        "_get_config",
+        return_value=[{"destination": "destination"}, {"datasets": ["a"]}],
+    ), patch.object(load, "create_temp_location", return_value=None), patch.object(
+        process, "process_dataset", return_value=tuple()
+    ) as patch_process_dataset, patch.object(
+        process,
+        "create_data_manifest",
+        return_value=pd.DataFrame({"id": ["a", "b", "c"]}),
+    ) as patch_create_data_manifest, patch.object(
+        load, "df_to_csv", return_value="path/to/csv"
+    ) as patch_df_to_csv, patch.object(
+        load, "load", return_value=None
+    ):
+        process.process_all_files(config_path=None, syn=syn)
+        patch_process_dataset.assert_called_once_with(dataset_obj="a", syn=syn)
+        patch_create_data_manifest.assert_called_once_with(
+            parent="destination", syn=syn
+        )
+        patch_df_to_csv.assert_called_once_with(
+            df=patch_create_data_manifest.return_value, filename="data_manifest.csv"
+        )
+
+
+def test_build_parser():
```

Comment:
Nit: I found something that could be better for testing argparsers: https://jugmac00.github.io/blog/testing-argparse-applications-the-better-way/#less-patching-more-passing-in.

Reply:
Doesn't this kind of un-do the purpose of patching? It seems like this is just testing the "other" way where you just pass a test case and see if the result is as expected. Not that my test is better - I don't really feel as though I am testing for much in my current version. Since our arg parser doesn't have any output to expect,  would we `assert` that the description is as expected? I don't see any obvious ways to test that the arguments are added correctly.

Reply:
You bring up a good point, I think this could be one usecase where we don't necessarily need to patch a python base functionality. 

That said, I am ok with leaving the test as is or potentially even removing it (but why not just leave it) for now because we run the test configuration during the last step of our CI/CD which tests the CLI.